iroh-resolver/src/resolver.rs

```diff
@@ -507,8 +510,20 @@ impl<T: ContentLoader> Resolver<T> {
 
                 *current = next_node;
             }
-            ty => {
-                bail!("unexpected unixfs type {:?}", ty);
+            UnixfsNode::HamtShard(node) => {
+                let hamt = Hamt::from_node(node)?;
+
+                let (next_link, next_node) = hamt
+                    .get(self, part.as_bytes())
+                    .await?
+                    .ok_or_else(|| anyhow!("link {} not found", part))?;
+                // TODO: is this the right way to to resolved path here?
+                resolved_path.push(next_link.cid);
+
+                *current = next_node.clone();
```

Comment:
Is there anything specific guarding us against infinite loops here. Ie is it possible to have a circular list of shards/nodes at all?

Reply:
the hamt itself has a recursion depth limit